### DISCUSSION TITLE
Sliding page transition with middleware. Is previous state being preserved?

### DISCUSSION THREAD

**Discussion by robokozo** [nuxt/nuxt#19757] [Questions]
I have a simple demo of a page transition effect I would like to achieve: https://stackblitz.com/edit/github-fzjp1w-yengqf?file=app.vue

Similar to what is seen here: https://nuxt.com/docs/getting-started/transitions#dynamic-transitions

Basically, I'd like to set the 'order' of the page in the meta data for the route and use that to correctly control the way in which the pages transition between each other. I'd love to contain the logic for determining which animation to run inside the `transition.js` middleware.

You'll notice that as you move from page 1 to page 2 - you don't get any animation at all. From 2 to 3 you get the appropriate slide effect. Going from 3 to 4 works as well.

Now the funky stuff begins when you start to change the direction. The first animation in the new direction (after going 1-2-3-4) and now going from 4 to 3 has the wrong exit animation. This can be confirmed by viewing the class attached to the element. 

Is this intentional? Perhaps a bug? Can anyone unwind or figure out how to fix this effect?

**Comment by tillderoquefeuil** [comment]
Hi there! I'm facing the same issue, did you sort that out?
<br>
**EDIT**
I was only defining the `to.meta.layoutTransition`, I had to apply the change on the `from.meta.layoutTransition` as well.
